server/routes/tasks.js

```diff
@@ -1,58 +1,70 @@
 export default (app) => {
   app
     .get('/task', async (req, reply) => {
-      const tasks = await app.objection.models.task.query().withGraphJoined('[status, creator, assigned]');
+      const tasks = await app.objection.models.task.query().withGraphJoined('[status, creator, assigned, labels]');
       await reply.render('task/index', { tasks });
     })
     .get('/task/new', async (req, reply) => {
       const statuses = await app.objection.models.status.query();
+      const labels = await app.objection.models.label.query();
       const users = await app.objection.models.user.query();
-      await reply.render('task/new', { data: { statuses, users }, errors: [] });
+      await reply.render('task/new', { data: { statuses, users, labels }, errors: [] });
     })
     .get('/task/edit/:id', async (req, reply) => {
       const task = await app.objection.models.task.query().findById(req.params.id);
       const statuses = await app.objection.models.status.query();
+      const labels = await app.objection.models.label.query();
       const users = await app.objection.models.user.query();
-      await reply.render('task/edit', { data: { statuses, users, task }, errors: [] });
+      await reply.render('task/edit', {
+        data: {
+          statuses, users, task, labels,
+        },
+        errors: [],
+      });
     })
     .post('/task', async (req, reply) => {
       try {
-        const { name, description } = req.body;
-        const statusId = Number(req.body.statusId);
-        const creatorId = Number(req.body.creatorId);
-        const assignedId = Number(req.body.assignedId);
-        const newTask = await app.objection.models.task.fromJson({
-          name,
-          description,
-          statusId,
-          creatorId,
-          assignedId,
+        await app.objection.models.task.query().insertGraph({
```

Comment:
insertGraph, updateGraph are not atomic operations. We should wrap them in transaction

Reply:
done